sr/comp/knockout_scheduler/static_scheduler.py

```diff
@@ -59,35 +72,39 @@ def get_team(self, team_ref: StaticMatchTeamReference | None) -> TLA | None:
             pos = int(team_ref[1:])
             # seed numbers are 1 based
             if pos < 1:
-                raise ValueError(f"Invalid seed {team_ref!r} (seed numbers start at 1)")
+                raise InvalidSeedError(f"Invalid seed {team_ref!r} (seed numbers start at 1)")
             pos -= 1
             try:
                 return self._knockout_seeds[pos]
             except IndexError:
-                raise ValueError(
+                raise InvalidSeedError(
                     "Cannot reference seed {}, there are only {} eligible teams!".format(
                         team_ref,
                         len(self._knockout_seeds),
                     ),
                 ) from None
 
         # get a position from a match
-        assert len(team_ref) == 3
-        round_num, match_num, pos = (int(x) for x in team_ref)
+        try:
+            round_num, match_num, pos = (int(x) for x in team_ref)
+        except ValueError:
+            raise InvalidReferenceError(
+                f"Match references must be three digits, not {team_ref!r}",
+            ) from None
 
         try:
             match = self.knockout_rounds[round_num][match_num]
         except IndexError:
-            raise ValueError(
-                f"Reference '{team_ref}' to unscheduled match!",
+            raise InvalidReferenceError(
+                f"Reference {team_ref!r} to unscheduled match!",
             ) from None
 
         try:
             ranking = self.get_ranking(match)
             return ranking[pos]
         except IndexError:
-            raise ValueError(
-                f"Reference '{team_ref}' to invalid ranking!",
+            raise InvalidReferenceError(
+                f"Reference {team_ref!r} to invalid ranking!",
```

Comment:
This is a slightly unclear error message (what's an invalid ranking?) - could do with (optional) clarification under the scout rule

Reply:
https://github.com/PeterJCLaw/srcomp/pull/36/commits/1123104ebaeccf2ef86c8843718f0108487b390e